### DISCUSSION TITLE
BigDecimal scalar returns numbers in responses, instead of strings

### DISCUSSION THREAD

**Discussion by Meemaw** [graphql-java/graphql-java#3692] [Q&A]
Hi,

We are using https://github.com/graphql-java/graphql-java-extended-scalars, and noticing that `BigDecimal` scalar is serializing things as numbers in the json response.

That can lead to precision loss for the clients. Fields that use the `BigDecimal` type should be returned as strings.

Is there some configuration to change that?

**Comment by bbakerman** [comment]
There is no configuration to make it return strings.

The serialize (output) implementation is

```
 public BigDecimal serialize(Object input) {
                BigDecimal result = convertImpl(input);
                if (result == null) {
                    throw new CoercingSerializeException(
                            "Expected type 'BigDecimal' but was '" + typeName(input) + "'."
                    );
                }
                return result;
            }
```

In other words the return value of the BigDecimal scalar is a BigDecimal.  So its precision loss is controlled by what can be represented by a BigDecimal.

I am not sure what examples you have in mind but it feels like a similar but different scalar.

**Comment by bbakerman** [comment]
> noticing that BigDecimal scalar is serializing things as numbers in the json response.

Also remember graphql-java never creates JSON - so your precision loss might be from your JSON encoder library of choice eg Jackson or GSon.
